### ISSUE TITLE
(Android) Nothing saves after exiting Retroarch

### ISSUE THREAD

**Issue by robobot2** [libretro/RetroArch#6996]
![screenshot_20180715-162752](https://user-images.githubusercontent.com/37221370/42738275-2412bb10-884f-11e8-9b23-4387de36e533.png)

# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

Nothing saves after exiting Retroarch app. Config, saves/states, etc.

### Expected behavior

Everything should save after exiting app.

### Actual behavior

Everything resets back to previous state. Save state stuck on 0% permanently.

### Bisect Results

Started happening once updated to Andorid 8.0.

### Version/Commit

Tried both currently nightly and latest stable from Playstore. Also tried wiping and doing a fresh install, but result is the same.

**Comment by fr500** [comment]
You changed your save dir from the default location right?

Sent from Mail for Windows 10

From: robobot2
Sent: Sunday, July 15, 2018 3:59 PM
To: libretro/RetroArch
Cc: Subscribed
Subject: [libretro/RetroArch] Nothing saves after exiting Retroarch (#6996)


First and foremost consider this:
• Only RetroArch bugs should be filed here. Not core bugs or game bugs
• This is not a forum or a help section, this is strictly developer oriented
Description
Nothing saves after exiting Retroarch app. Config, saves/states, etc.
Expected behavior
Everything should save after exiting app.
Actual behavior
Everything resets back to previous state. Save state stuck on 0% permanently.
Bisect Results
Started happening once updated to Andorid 8.0.
Version/Commit
Tried both currently nightly and latest stable from Playstore. Also tried wiping and doing a fresh install, but result is the same.

**Comment by robobot2** [comment]
Yes, but not the config file. That's in the default location in internal storage, but that one doesn't save either.

Saves/states are in SD card. Either way, nothing works.

**Comment by fr500** [comment]
Saves on the SD card won't work.
Google won't allow android apps to write there.

If the config doesn't save you're probably closing by swiping.
Try closing via the menu entry.

**Comment by robobot2** [comment]
No. I'm closing properly through the menu. I already know about exiting through the menu properly for options to stay saved and I also have the option enabled, so there shouldn't be a problem.

And it has nothing to do with whether it's on SD card or not because I tried testing it by putting the saves/states in internal. The result is the same.

**Comment by fr500** [comment]
Is the config saving or not?
I don't have the issue in my XPERIA XZ1 running Oreo

**Comment by robobot2** [comment]
It's not. Neither are the saves even when they're in internal. Nothing is being saved.

**Comment by fr500** [comment]
Hmmm, well there is nothing we can do without a log, the device name, type, etc.
If it started once you updated to android 8.0 then something change permissions-wise in the rom, nothing has changed over here in that department on a couple of years.

**Comment by robobot2** [comment]
I'm thinking it has something do with permission as well, but I have all of it enabled. How do I make a log?

My device is Moto Z Play running 8.0.

**Comment by fr500** [comment]
adb logcat
https://www.howtogeek.com/125769/how-to-install-and-use-abd-the-android-debug-bridge-utility/

**Comment by fr500** [comment]
Sorry that isn't useful at all, you should logcat from PC and get a full file
you can also do 

adb shell
and inside the shell
logcat | grep RetroArch

to get relevant information only

**Comment by robobot2** [comment]
How do I get the full file?

**Comment by bparker06** [comment]
>RetroArch: Saving RAM type #0 to "/storage/3030
3030-2d30-3030-3000-000000040000/retroarch/Save/Pokemon Emerald.srm".

Looks like it is trying to save to SD or some external storage. Given that we know this definitely won't work, we'd want to see a log where it's failing to save to internal storage instead.

**Comment by fr500** [comment]
And it looks like the config saved just fine:

```
[Config]: Saved new config to "/stor
age/emulated/0/Android/data/com.retroarch/files/retroarch.cfg
```

**Comment by robobot2** [comment]
"Looks like it is trying to save to SD or some external storage. Given that we know this definitely won't work, we'd want to see a log where it's failing to save to internal storage instead."

It's strange, I set it up in the config to save to internal, but it still shows up as SD card location. Maybe that's the problem?

"And it looks like the config saved just fine"

But the config remains the same though...the options don't stick. For example, I changed the location  of the save to internal, exited Retroarch through the menu, but when I open it again, it defaults to the SD card location.
